src/pages/_app.tsx

```diff
@@ -0,0 +1,48 @@
+import { initializeIcons } from '@fluentui/react'
+import { AppContext, AppInitialProps, AppProps } from 'next/app'
+import { useEffect } from 'react'
+import { StaticRouter, BrowserRouter } from 'react-router-dom'
+import 'theme/normalize.scss'
+
+export const ReactRouterContext = Symbol('React Router Context')
+
+initializeIcons()
+
+export default function App(props: AppProps) {
+  const { Component, pageProps, router } = props
+  useEffect(() => {
+    // disable nextjs's router: react-router will handle page transitions instead:
+    router.beforePopState(() => {
+      return false
+    })
+  }, [])
+
+  if (typeof window === 'undefined') {
+    return (
+      <StaticRouter
+        location={router.asPath}
+        context={(router.query as any)[ReactRouterContext]}
+      >
+        <Component {...pageProps} />
+      </StaticRouter>
+    )
+  } else {
+    return (
+      <BrowserRouter>
+        <Component {...pageProps} />
+      </BrowserRouter>
+    )
+  }
+}
+
+App.getInitialProps = async (
+  appContext: AppContext
+): Promise<AppInitialProps> => {
+  // Specify getInitialProps so nextjs populates the router.asPath with the
+  // actual request path. Then, create the react-router context in the router
+  // query object (which is the only object here also accessible in  _document)
+  ;(appContext.router.query as any)[ReactRouterContext] = {}
```

Comment:
nit: the semicolon seems misplaced, is this the linter?

Reply:
yeah I think this was the linter as I didn't modify this manually...

Reply:
https://github.com/prettier/prettier/issues/1935#issuecomment-306729468 this seems to be prettier doing it

Reply:
Updated prettier to `semi: true` to match central and remove this issue